ee/vellum_ee/workflows/display/tests/workflow_serialization/test_basic_inline_prompt_node_serialization.py

```diff
@@ -274,3 +274,49 @@ def test_serialize_workflow():
         "name": "BasicInlinePromptWithFunctionsWorkflow",
         "module": ["tests", "workflows", "basic_inline_prompt_node_with_functions", "workflow"],
     }
+
+
+def test_serialize_workflow_with_descriptor_functions():
+    """Test that serialization handles BaseDescriptor instances in functions list."""
+    from vellum import ChatMessagePromptBlock, JinjaPromptBlock
+    from vellum.workflows import BaseWorkflow
+    from vellum.workflows.inputs import BaseInputs
+    from vellum.workflows.nodes import InlinePromptNode
+    from vellum.workflows.state import BaseState
+    from vellum_ee.workflows.display.workflows.get_vellum_workflow_display_class import get_workflow_display
+
+    class TestInputs(BaseInputs):
+        noun: str
+
+    from vellum.workflows.references.node import NodeReference
+
+    class MockMCPClientNode:
+        class Outputs:
+            tools = NodeReference(
+                name="tools",
+                types=(list,),
+                instance=["tools.0", "tools.1"],  # Non-callable objects that will cause the error
+                node_class=object,
+            )
```

Comment:
make `tools` a list of valid function definitions. I promise the error will still reproduce bc of the association below